### PR TITLE
CL-31 As a user, I want to be able to login

### PR DESCRIPTION
**Pull request by stevelw**

### CODE REVIEW

**Comment by stevelw** [comment]
Should base on #20 after commits added to branch

File: src/network.ts [hunk 1]
```diff
@@ -31,3 +31,19 @@ export function updateUser(data: User): Promise<User> {
 			return {};
 		});
 }
+
+export function registerUser(
+	username: string
+): Promise<null | { username: string }> {
+	return network
+		.post('/users', {
```
Comment (jeepies) [inline src/network.ts:39]: Incorrectly formed URL. All the others use /api/[x]

File: src/Components/Login.tsx [hunk 2]
```diff
@@ -1,13 +1,39 @@
...
+
+		return registerUser(formUsername).then(
+			(newUser: null | { username: string }) => {
+				if (!newUser) {
+					setErrorMessage(
+						'Sorry, there was an error creating an account. Please try again later.'
+					);
+				} else {
+					setUsername(newUser.username);
+					setHome([parseFloat(latitude), parseFloat(longitude)]);
```
Comment (jeepies) [inline src/Components/Login.tsx:33]: Nothing happens after this - I wanted to go to the map , Steve! :(

File: src/Components/Login.tsx [hunk 3]
```diff
@@ -1,13 +1,39 @@
...
+
+	function isValidLat(input: string): boolean {
+		const value = parseFloat(input);
+		return !isNaN(value) && value >= -90 && value <= 90;
+	}
+
+	function isValidLong(input: string): boolean {
+		const value = parseFloat(input);
+		return !isNaN(value) && value >= -180 && value <= 180;
+	}
```
Comment (jeepies) [inline src/Components/Login.tsx:20]: We need some error messages for this! I put in `aaaaaa` and was confused because there was no message :b

**Review by jeepies — CHANGES_REQUESTED** [review]

File: src/Components/Login.tsx (hunk 2 above)
Comment (stevelw) [reply src/Components/Login.tsx:33]: I assume that's because of the wrong endpoint url?

File: src/Components/Login.tsx (hunk 3 above)
Comment (stevelw) [reply src/Components/Login.tsx:20]: OK, will work on tomorrow

Comment (stevelw) [reply src/Components/Login.tsx:23]: Is this OK now or did you need more?

Comment (jeepies) [reply src/Components/Login.tsx:23]: For an MVP sure

File: src/Components/Login.tsx (hunk 2 above)
Comment (jeepies) [reply src/Components/Login.tsx:36]: Nope - It 204's now, content created, and doesnt redirect.

**Review by mizamidou — APPROVED** [review]

**Comment by stevelw** [comment]
Throwing on backlog

**Pull request closed without merging**